controller/backing_image_manager_controller.go

```diff
@@ -388,6 +391,40 @@ func (c *BackingImageManagerController) cleanupBackingImageManager(bim *longhorn
 	return nil
 }
 
+// evictMissingDiskBackingImageManager trigger image manager eviction for missing disks
+func (c *BackingImageManagerController) evictMissingDiskBackingImageManager(bim *longhorn.BackingImageManager) error {
+	node, _ := c.ds.GetNode(bim.Spec.NodeID)
+	if node == nil {
+		return nil
+	}
+	for _, diskStatus := range node.Status.DiskStatus {
+		if diskStatus.DiskUUID == bim.Spec.DiskUUID {
+			return nil
+		}
+	}
+
+	// Backing image manager is serving on the disk that no longer belongs to any node. Trigger the manager eviction.
+	for imageName := range bim.Spec.BackingImages {
+		bi, getImageErr := c.ds.GetBackingImageRO(imageName)
+		if getImageErr != nil {
+			if datastore.ErrorIsNotFound(getImageErr) {
+				c.logger.Warnf("No corresponding backing image %v for missing disk backing image manager %v", imageName, bim.Name)
+				continue
+			} else {
+				return errors.Wrapf(getImageErr, "Failed to get backing image %v for missing disk backing image manager %v", bi.Name, bim.Name)
+			}
+		}
+		if bimDiskFileSpec, exist := bi.Spec.DiskFileSpecMap[bim.Spec.DiskUUID]; exist {
```

Comment:
Is it better to check if `DiskFileSpecMap` is not `nil` before accessing `bi.Spec.DiskFileSpecMap[bim.Spec.DiskUUID]`?